### ISSUE TITLE
Only touch the fsproj when things actually changed

### ISSUE THREAD

**Issue by forki** [fsprojects/Mechanic#83]
![image](https://user-images.githubusercontent.com/57396/37339531-3f5e7684-26bb-11e8-97cf-af3adef85d63.png)

we should only overwrite the file if the content changed

**Comment by dpraimeyuu** [comment]
Hi @forki :-)

I am newcomer to F# OSS and I would like to help. As I understood, this issue is about saving current fsproj file if any file ordering was applied, am I correct? If so, can this be detected purely using content of fsproj?

**Comment by forki** [comment]
yes exactly. I paket we basically load the file into memory and compare with string that would write to disk. If the strings are equal then we skip the writing

**Comment by dpraimeyuu** [comment]
Doesn't seem scary though. I will try to do setup this evening

**Comment by dpraimeyuu** [comment]
@forki 

![image](https://user-images.githubusercontent.com/6437191/37446066-802e605a-281b-11e8-8e4b-ed54cea17889.png)

I suppose here (before invoking ```save``` fn) the actual checking should be placed, correct?

**Comment by forki** [comment]
Yes. Or inside the save function

Damian Płaza <notifications@github.com> schrieb am Do., 15. März 2018,
06:39:

**Comment by dpraimeyuu** [comment]
Then ```save``` fn would do two things: check if save is necessary and true saving. I would place it in mentioned ```updateProjectFile```.

**Comment by forki** [comment]
Ok

Damian Płaza <notifications@github.com> schrieb am Do., 15. März 2018,
06:52:
